main/src/main/java/cgeo/geocaching/activity/AbstractActionBarActivity.java

```diff
@@ -1,20 +1,62 @@
 package cgeo.geocaching.activity;
 
+import cgeo.geocaching.R;
+import cgeo.geocaching.ui.ViewUtils;
+import cgeo.geocaching.utils.Log;
+
 import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.ActionBar;
+import androidx.core.graphics.Insets;
+import androidx.core.view.OnApplyWindowInsetsListener;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowCompat;
+import androidx.core.view.WindowInsetsCompat;
+import androidx.core.view.WindowInsetsControllerCompat;
 
 /**
  * Classes actually having an ActionBar (as opposed to the Dialog activities)
  */
 public class AbstractActionBarActivity extends AbstractActivity {
 
+    protected int calculateInsets = WindowInsetsCompat.Type.statusBars() | WindowInsetsCompat.Type.displayCutout();
+    boolean skipActionBarInsetCalculation = false; // for edge to edge configuration, see onCreate()
+
     @Override
     public void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         initUpAction();
-    }
 
+        // edge2edge configuration
+        final Window currentWindow = getWindow();
```

Comment:
Some of this logic might be needed in other places as well (e.g. for dialogs). It might make sense to move it to a utils class e.g. ViewUitls?

Reply:
I had in mind eg the full-screen dialigs used in Wherigo Player. They also beed edge-to-edge enabled but are not activity-based. But we can do that in a later step/PR